docs/can-guides/contribute/naming-conventions.md

```diff
@@ -0,0 +1,46 @@
+@page guides/contributing/naming-conventions Naming Conventions
+@parent guides/contribute
+
+@description Learn how to design APIs which are consistent with CanJS.
+
+@body
+
+## Overview
+
+CanJS has many goals, one of which is providing a consistent interface and experience for developers using similar functionality or usage patterns across projects. Ideally, a developer should have a rough understanding of how to use something new based on what they have already used in CanJS.
+
+This document is geared towards contributors who are designing new APIs for CanJS and developers of libraries meant to work alongside CanJS projects. Below are useful conventions for designing APIs, which have been adopted by CanJS projects or will be aligned with in future releases.
+
+## Conventions
+
+### Variable casing
+
+All methods and properties should be camelCase. Class and constructor function names are exclusively PascalCase. 
+
+### Methods are verbs, properties are nouns
+
+A developer should be able to tell by looking at an identifier whether it is a function that can be called or an object/property that has a value. In JavaScript, functions can be values so be careful to make the distinction between a *method* (verb) and a function being passed around as data (noun). For example, `handleEvent` is a method, but if that method were being passed around `eventHandler` is more appropriate.
+
+Methods which are only verbs such as `jump()` are discouraged when it is not clear what should be passed to it. A better name `jumpOverPerson()` provides a hint as to what the method expects.
+
+### Private methods and variables
+
+Any method or variable beginning with one or more underscores (`_`) is private to the project and should not be used externally. Ideally these cases are minimized.
```

Comment:
I think we should have `_KEY` as private, but document `__KEY` as NOT necessarily private.  For example the [__keys](https://canjs.com/doc/can-define/map/map/KeysEvent.html) event can be bound to, but is named `__keys` to prevent collisions with other properties.  Symbols can eventually fix this, but it's how we avoid collisions currently.